subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/IncrementalCompilerDecorator.java

```diff
@@ -83,4 +95,52 @@ public IncrementalCompilerDecorator(JarClasspathSnapshotMaker jarClasspathSnapsh
         PreviousCompilation previousCompilation = new PreviousCompilation(new ClassSetAnalysis(data), compileCaches.getLocalJarClasspathSnapshotStore(), compileCaches.getJarSnapshotCache());
         return new SelectiveCompiler(inputs, previousCompilation, cleaningCompiler, staleClassDetecter, compilationInitializer, jarClasspathSnapshotMaker);
     }
+
+    // Check whether there are any Annotation Processors on the classpath that are NOT
+    // incap-enabled (i.e. incremental).  If so, then we do not allow an incremental build.
+    // If there is no annotation processing at all, then we permit an incremental build.
+    private boolean allowIncrementalAnnotationProcessing() {
+        if (!compileOptions.isIncrementalAnnotationProcessing()) {
+            return false;
+        }
+        FileCollection classPath = new SimpleFileCollection(javaCompileSpec.getCompileClasspath());
+        Set<AnnotationProcessorInfo> cachedInfo = annotationProcessorDetector.getAnnotationProcessorInfo(compileOptions, classPath);
+        if (annotationProcessorDetector.checkExplicitProcessorOption(compileOptions)) {
+            LOG.info("{} is not incremental:  The explicit -processor compiler option is not supported for incremental annotation processing.", displayName);
+            return false;
+        }
+        boolean foundProcessors = false;
+        Set<String> badProcs = Sets.newHashSet();
+        for (AnnotationProcessorInfo info : cachedInfo) {
+            if (info.isProcessor()) {
+                foundProcessors = true;
+                if (!info.isIncrementalEnabled()) {
+                    badProcs.add(info.getName());
+                }
+            }
+        }
+        if (!badProcs.isEmpty()) {
+            printBadProcessors(badProcs);
+            return false;
+        }
+        if (foundProcessors) {
+            LOG.warn("{} - all annotation processors are incremental.", displayName);
```

Comment:
❌ All these messages should be info. Also, we don't need a message if everything is okay (i.e. incremental). We only need a message if it isn't.

Reply:
Oops, that was for debugging.  Fixed.